platform/posix/transport/src/openssl_posix.c

```diff
@@ -623,10 +625,15 @@ int32_t Openssl_Recv( NetworkContext_t * pNetworkContext,
         }
         else
         {
-            LogError( ( "SSL_read of OpenSSL failed to receive data: "
-                        "status=%s.", ERR_reason_error_string( sslError ) ) );
+            LogError( ( "Failed to receive data over network: SSL_read failed: "
+                        "ErrorStatus=%s.", ERR_reason_error_string( sslError ) ) );
         }
     }
+    else if( pNetworkContext->pSsl == NULL )
```

Comment:
In Recv, we use `else if` statement to check if pNetworkContext->pSsl is `NULL` but below, we use an `else` statement. I believe the latter is easier to read, but it is your call. I am just saying we should be consistent about how the logic is being written.

Reply:
The error handling logic of `Recv` function is different from that of the `Send` function which is why I had used `else-if`
However, have re-adjusted some logic to make the code simpler.